plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1047 @@
+            - You need to set this password on the lxd server before
+                running this module using the following command
+                C(lxc config set core.trust_password <some random password>)
+                See U(https://www.stgraber.org/2016/04/18/lxd-api-direct-interaction/).
+            - If I(trust_password) is set, this module send a request for authentication before sending any requests.
+            type: str
+        state:
+            description: Filter the container according to the current status.
+            type: str
+            default: none
+            choices: [ 'STOPPED', 'STARTING', 'RUNNING', 'none' ]
+        prefered_container_network_interface:
+            description:
+            - If a container has multiple network interfaces, select which one is the prefered as pattern.
+            - Combined with the first number that can be found e.g. 'eth' + 0.
+            type: str
+            default: eth
+        prefered_container_network_family:
+            description:
+            - If a container has multiple network interfaces, which one is the prefered by family.
+            - Specify C(inet) for IPv4 and C(inet6) for IPv6.
+            type: str
+            default: inet
+            choices: [ 'inet', 'inet6' ]
+        dispose:
+            description:
+            - Create groups by the following keywords C(location), C(pattern), C(network_range), C(os), C(release), C(profile), C(vlanid).
+            - See example for syntax.
+            type: json
+'''
+
+EXAMPLES = '''
+# simple lxd.yml
+plugin: community.general.lxd
+url: unix:/var/snap/lxd/common/lxd/unix.socket
+
+# simple lxd.yml including filter
+plugin: community.general.lxd
+url: unix:/var/snap/lxd/common/lxd/unix.socket
+state: RUNNING
+
+# grouping lxd.yml
+dispose:
+  testpattern:
+    type: pattern
+    attribute: test
+  vlan666:
+    type: vlanid
+    attribute: 666
+  locationBerlin:
+    type: location
+    attribute: Berlin
+  osUbuntu:
+    type: os
+    attribute: ubuntu
+  releaseFocal:
+    type: release
+    attribute: focal
+  releaseBionic:
+    type: release
+    attribute: bionic
+  profileDefault:
+    type: profile
+    attribute: default
+  profileX11:
+    type: profile
+    attribute: x11
+  netRangeIPv4:
+    type: network_range
+    attribute: 10.98.143.0/24
+  netRangeIPv6:
+    type: network_range
+    attribute: fd42:bd00:7b11:2167:216:3eff::/24
+'''
+
+import binascii
+import json
+import re
+import time
+import os
+import socket
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common.dict_transformations import dict_merge
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible_collections.community.general.plugins.module_utils.lxd import LXDClient, LXDClientException
+
+
+class InventoryModule(BaseInventoryPlugin):
+    DEBUG = 4
+    NAME = 'community.general.lxd'
```

Comment:
Given we have a number other resources named `lxd*` inside c.g., I would suggest we were more explicit in the naming here, something like `lxd_inventory`. I am aware that other inventory plugins do not follow that concept, so I will not press this point further for consistency sake.

Reply:
I disagree on this one: this is already determined by the plugin type, there's absolutely no reason to make users type more by adding the plugin type to the FQCN.